client.go

```diff
@@ -41,6 +41,13 @@ var (
 	regexpIllegalNameChars = regexp.MustCompile("[^A-Za-z0-9.:_-]") // from https://www.AppOptics.com/docs/api/#measurements
 	// ErrBadStatus is returned if the AppOptics API returns a non-200 error code.
 	ErrBadStatus = errors.New("Received non-OK status from AppOptics POST")
+	client = &http.Client{
+		Timeout: 30 * time.Second,
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: 4,
+			IdleConnTimeout:     30 * time.Second,
+		},
```

Comment:
This sort of surprised me because the Go stdlib often behaves sanely with the zero value of config parameters, but by defining a custom `http.Transport`, this loses some of the sanity-preserving defaults in the `DefaultTransport`, which you can see in https://golang.org/pkg/net/http/#RoundTripper. For example, DefaultTransport defines `MaxIdleConns: 100`, while the custom Transport here gets a default of 0, and the docs for Transport say "Zero means no limit," which is probably not desirable though may not have immediate impact either.

What's the reason for `MaxIdleConnsPerHost`? The default is `2` (i.e. `http.DefaultMaxIdleConnsPerHost`). Did we run out of ports before or something? Ditto with `IdleConnTimeout`. I kind of wonder if this `Transport` declaration should be removed now that we identified the cause of not reusing connections?

Reply:
I would be in favor of going back to the default transport unless we have a specific reason for these overrides, @trevrosen.